### PR TITLE
Add acceptance test to accept previously declined share and vice-versa

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance test to accept previously declined share, and decline an accepted share

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 1]
```diff
@@ -312,3 +299,87 @@ Feature: accept/decline shares coming from internal users
...
+    Given the setting "shareapi_auto_accept_share" in the "core" section has been set to "no"
+    And user "user1" has shared file "lorem.txt" with user "user2"
+    When the user browses to the shared-with-me page using the webUI
+    Then the file "lorem.txt" on shared-with-me page should be in "Pending" state on the webUI
+    When the user browses to the files page
+    Then file "lorem (2).txt" should not be listed on the webUI
+
+  Scenario: receive shares with same name from different users
+    Given user "user3" has been created with default attributes
+    Given the setting "shareapi_auto_accept_share" in the "core" section has been set to "no"
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]:
```suggestion
    And the setting "shareapi_auto_accept_share" in the "core" section has been set to "no"
```

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 2]
```diff
@@ -312,3 +299,87 @@ Feature: accept/decline shares coming from internal users
...
+    When the user browses to the shared-with-me page using the webUI
+    Then file "lorem.txt" shared by "User One" should be in state "Pending" in the shared-with-me page on the webUI
+    And file "lorem.txt" shared by "User Three" should be in state "Pending" in the shared-with-me page on the webUI
+
+  Scenario: decline an offered (pending) share
+    Given the setting "shareapi_auto_accept_share" in the "core" section has been set to "no"
+    And user "user1" has shared file "lorem.txt" with user "user2"
+    And user "user1" has shared file "testimage.jpg" with user "user2"
+    And the user has browsed to the shared-with-me page
+    When the user "Decline" share "lorem.txt" offered by user "User One" using the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: can we do `declines` here? so that is sounds more natural?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/5923/

<p>

























</p></details>

**Comment by jasson99** [comment]
@individual-it @phil-davis

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 3]
```diff
@@ -21,11 +21,10 @@ module.exports = {
...
      * declined or pending
      */
-    assertDesiredStatusIsPresent: function (filename, status, user) {
+    assertDesiredStatusIsPresent: function (filename, status, user = undefined) {
       let requiredXpath = this.api.page.FilesPageElement.filesList().getFileRowSelectorByFileName(filename) +
-        util.format(this.elements.assertStatusFileRow.selector, status)
-      requiredXpath = user === undefined ? requiredXpath : requiredXpath +
-                      util.format(this.elements.getSharedFromUserName.selector, user)
+                          util.format(this.elements.assertStatusFileRow.selector, status)
+      requiredXpath = user !== undefined ? requiredXpath + util.format(this.elements.getSharedFromUserName.selector, user) : requiredXpath
```
Comment (phil-davis) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: why do these lines need to be changed?
IMO the logic has been sswapped around, but it ends up the same behavior.

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 4]
```diff
@@ -464,6 +464,12 @@ Then('the file {string} should be in {string} state on the webUI after a page re
   await client.refresh()
   return client.page.sharedWithMePage().assertDesiredStatusIsPresent(filename, status)
 })
+
 Then('the autocomplete list should not be displayed on the webUI', function () {
   return client.page.FilesPageElement.sharingDialog().assertAutocompleteListIsNotVisible()
 })
+
+Given('user {string} has declined the share {string} offered by user {string}', async function (user, filename, sharer) {
+  await sharingHelper.getAllShares(user)
```
Comment (phil-davis) [inline tests/acceptance/stepDefinitions/sharingContext.js]: why is this needed?
`declineShare` already calls `getAllShares` inside itself.

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 5]
```diff
@@ -349,3 +349,45 @@ Feature: accept/decline shares coming from internal users
...
+    And user "user2" has declined the share "lorem.txt" offered by user "user1"
+    And the user has browsed to the shared-with-me page
+    When the user accepts share "lorem.txt" offered by user "User One" using the webUI
+    Then the file "lorem (2).txt" should be in "Accepted" state on the webUI
+    And the file "testimage.jpg" should be in "Pending" state on the webUI
+    When the user browses to the files page
+    Then file "lorem (2).txt" should be listed on the webUI
+    And file "testimage (2).jpg" should not be listed on the webUI
+
+  Scenario: decline an accepted share (with page-reload in between)
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]:
```suggestion
  Scenario: delete an accepted share (with page-reload in between)
```

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature [hunk 6]
```diff
@@ -349,3 +349,45 @@ Feature: accept/decline shares coming from internal users
...
+    And the user reloads the current page of the webUI
+    And the user browses to the files page
+    And the user deletes file "lorem (2).txt" using the webUI
+    Then file "lorem (2).txt" should not be listed on the webUI
+    And file "testimage (2).jpg" should not be listed on the webUI
+    When the user browses to the shared-with-me page using the webUI
+    Then the file "lorem.txt" should be in "Declined" state on the webUI
+    And the file "testimage.jpg" should be in "Pending" state on the webUI
+
+  Scenario: decline an accepted share (without page-reload in between)
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]:
```suggestion
  Scenario: delete an accepted share (without page-reload in between)
```

File: tests/acceptance/helpers/sharingHelper.js [hunk 7]
```diff
@@ -133,6 +133,40 @@ module.exports = {
...
+    return fetch(apiURL,
+      {
+        method: 'GET',
+        headers: headers
+      })
+      .then(res => {
+        httpHelper.checkStatus(res, 'The response status is not the expected value')
+        return res.json()
+      })
+      .then((res) => {
```
Comment (HariBhandari07) [inline tests/acceptance/helpers/sharingHelper.js]: ```.then((res) => {```
can be ```.then( res => {``` 
but it's not an issue, just a suggestion.

File: tests/acceptance/helpers/sharingHelper.js [hunk 8]
```diff
@@ -133,6 +133,40 @@ module.exports = {
...
+        return res.ocs.data
+      })
+  },
+  declineShare: async function (filename, user, sharer) {
+    const allShares = await this.getAllShares(user)
+    console.log(allShares)
```
Comment (HariBhandari07) [inline tests/acceptance/helpers/sharingHelper.js]: remove this debug line

File: tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature (hunk 5 above)
Comment (jasson99) [reply tests/acceptance/features/webUISharingAcceptShares/acceptShares.feature]: the main purpose of deleting here was to decline the accepted share..Moreover, I implemented this the same way as in the core(i guess so because these scenarios were written already but skipped). Would it be okay if I changed "decline" to "delete" ?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6041/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6100/

<p>










</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6119/

<p>













</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6119/

<p>






































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6122/

<p>

































</p></details>

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 9]
```diff
@@ -53,6 +53,10 @@ module.exports = {
     }
   },
   elements: {
+    pendingStatusFileRow: {
```
Comment (dpakach) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: I dont see this being used anywhere.

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 10]
```diff
@@ -64,6 +68,10 @@ module.exports = {
     actionOnFileRow: {
       selector: '/../..//a[.="%s"]',
       locateStrategy: 'xpath'
+    },
+    fileStatusOfFileRow: {
```
Comment (dpakach) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: Same Here not used anywhere.

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingPublic</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6160/

<p>

































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6160/

<p>





























</p></details>
